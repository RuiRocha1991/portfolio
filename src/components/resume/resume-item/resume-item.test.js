import React from 'react'
import {configure,  shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer'

configure({ adapter: new Adapter() });

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faGraduationCap, faSuitcase } from '@fortawesome/free-solid-svg-icons'

import ResumeItem from '.'

it('Should ResumeItem with all props and isGraduation=true to match snapshot',() => {
  const tree = renderer.create(
    <ResumeItem
      title={"Computer Science"}
      startDate={2007}
      endDate={2011}
      affiliations={"Harvard University"}
      resume={"Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusm tempor incididunt ut labore et dolore magna aliqua Lorem ipsum dolor sit amet, consectetur adipisicing elit."}
      isGraduation={true}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Should ResumeItem with all props and isGraduation=true contains icon "faGraduationCap"',() => {
  const wrapper = shallow(<ResumeItem
    title={"Computer Science"}
    startDate={2007}
    endDate={2011}
    affiliations={"Harvard University"}
    resume={"Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusm tempor incididunt ut labore et dolore magna aliqua Lorem ipsum dolor sit amet, consectetur adipisicing elit."}
    isGraduation={true}
  />)
  const icon = <FontAwesomeIcon className="resume-item-icon" icon={faGraduationCap}/>
  expect(wrapper.contains(icon)).toEqual(true);
})

it('Should ResumeItem with all props and isGraduation=false to match snapshot',() => {
  const tree = renderer.create(
    <ResumeItem
      title={"Computer Science"}
      startDate={2007}
      endDate={2011}
      affiliations={"Harvard University"}
      resume={"Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusm tempor incididunt ut labore et dolore magna aliqua Lorem ipsum dolor sit amet, consectetur adipisicing elit."}
      isGraduation={false}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Should ResumeItem with all props and isGraduation=false contains icon "faSuitcase"',() => {
  const wrapper = shallow(<ResumeItem
    title={"Computer Science"}
    startDate={2007}
    endDate={2011}
    affiliations={"Harvard University"}
    resume={"Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusm tempor incididunt ut labore et dolore magna aliqua Lorem ipsum dolor sit amet, consectetur adipisicing elit."}
    isGraduation={false}
  />)
  const icon = <FontAwesomeIcon className="resume-item-icon" icon={faSuitcase}/>
  expect(wrapper.contains(icon)).toEqual(true);
})

it('Should ResumeItem without props to match snapshot',() => {
  const tree = renderer.create(
    <ResumeItem/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})


it('Should ResumeItem validate title',() => {
  const wrapper = shallow(<ResumeItem
    title={"Computer Science"}
    startDate={2007}
    endDate={2011}
    affiliations={"Harvard University"}
    resume={"Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusm tempor incididunt ut labore et dolore magna aliqua Lorem ipsum dolor sit amet, consectetur adipisicing elit."}
    isGraduation={false}
  />)
  const title =  <h4>Computer Science</h4>
  expect(wrapper.contains(title)).toEqual(true);
})
