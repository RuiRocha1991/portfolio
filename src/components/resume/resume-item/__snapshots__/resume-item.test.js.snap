// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should ResumeItem with all props and isGraduation=false to match snapshot 1`] = `
<div
  className="resume-item-container"
>
  <div
    className="resume-item"
  >
    <div
      className="resume-item-icon-container-experience"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-suitcase fa-w-16 resume-item-icon"
        data-icon="suitcase"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128 480h256V80c0-26.5-21.5-48-48-48H176c-26.5 0-48 21.5-48 48v400zm64-384h128v32H192V96zm320 80v256c0 26.5-21.5 48-48 48h-48V128h48c26.5 0 48 21.5 48 48zM96 480H48c-26.5 0-48-21.5-48-48V176c0-26.5 21.5-48 48-48h48v352z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="resume-date"
    >
      2007
       - 
      2011
    </div>
    <h4>
      Computer Science
    </h4>
    <span>
      Harvard University
    </span>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusm tempor incididunt ut labore et dolore magna aliqua Lorem ipsum dolor sit amet, consectetur adipisicing elit.
    </p>
  </div>
</div>
`;

exports[`Should ResumeItem with all props and isGraduation=true to match snapshot 1`] = `
<div
  className="resume-item-container"
>
  <div
    className="resume-item"
  >
    <div
      className="resume-item-icon-container-graduation"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-graduation-cap fa-w-20 resume-item-icon"
        data-icon="graduation-cap"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M622.34 153.2L343.4 67.5c-15.2-4.67-31.6-4.67-46.79 0L17.66 153.2c-23.54 7.23-23.54 38.36 0 45.59l48.63 14.94c-10.67 13.19-17.23 29.28-17.88 46.9C38.78 266.15 32 276.11 32 288c0 10.78 5.68 19.85 13.86 25.65L20.33 428.53C18.11 438.52 25.71 448 35.94 448h56.11c10.24 0 17.84-9.48 15.62-19.47L82.14 313.65C90.32 307.85 96 298.78 96 288c0-11.57-6.47-21.25-15.66-26.87.76-15.02 8.44-28.3 20.69-36.72L296.6 284.5c9.06 2.78 26.44 6.25 46.79 0l278.95-85.7c23.55-7.24 23.55-38.36 0-45.6zM352.79 315.09c-28.53 8.76-52.84 3.92-65.59 0l-145.02-44.55L128 384c0 35.35 85.96 64 192 64s192-28.65 192-64l-14.18-113.47-145.03 44.56z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="resume-date"
    >
      2007
       - 
      2011
    </div>
    <h4>
      Computer Science
    </h4>
    <span>
      Harvard University
    </span>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusm tempor incididunt ut labore et dolore magna aliqua Lorem ipsum dolor sit amet, consectetur adipisicing elit.
    </p>
  </div>
</div>
`;

exports[`Should ResumeItem without props to match snapshot 1`] = `
<div
  className="resume-item-container"
>
  <div
    className="resume-item"
  >
    <div
      className="resume-item-icon-container-experience"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-suitcase fa-w-16 resume-item-icon"
        data-icon="suitcase"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M128 480h256V80c0-26.5-21.5-48-48-48H176c-26.5 0-48 21.5-48 48v400zm64-384h128v32H192V96zm320 80v256c0 26.5-21.5 48-48 48h-48V128h48c26.5 0 48 21.5 48 48zM96 480H48c-26.5 0-48-21.5-48-48V176c0-26.5 21.5-48 48-48h48v352z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="resume-date"
    >
      0
       - 
      0
    </div>
    <h4>
      N/A
    </h4>
    <span>
      N/A
    </span>
    <p>
      N/A
    </p>
  </div>
</div>
`;
